src/main/kotlin/org/xyro/kumulus/KumulusTopology.kt

```diff
@@ -290,26 +309,50 @@ class KumulusTopology(
             }
         } else {
             logger.trace { "Component ${c.componentId}/${c.taskId} is currently busy" }
-            if (onBusyBoltHook != null && message is ExecuteMessage) {
+            var shouldEnqueue = true
+            if (message is ExecuteMessage) {
+                val messageWaitStartTime =  c.waitStart.get()
+
+                if (messageWaitStartTime > 0){
+                    if ((System.nanoTime() - messageWaitStartTime >= lateMessageMaxWaitInNanos) &&
+                        lateMessagesStreamsToDrop.contains(message.tuple.kTuple.sourceStreamId)) {
+                        if (!message.isLate.get()){
+                            atomicNumberOfMessageThatShouldBeDropped.incrementAndGet()
```

Comment:
`atomicNumberOfMessageThatShouldBeDropped` is only used for test purposes. Can we maybe nullify it it production runs to prevent another atomic operation?